catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/QueryFilterTransformerProvider.java

```diff
@@ -47,32 +52,64 @@ public synchronized void bind(ServiceReference<QueryFilterTransformer> reference
 
     for (QName namespace : namespaces) {
       queryFilterTransformerMap.put(namespace, transformer);
+      List<String> typeNames = getTypeNames(reference);
+      if (CollectionUtils.isNotEmpty(typeNames)) {
+        for (String typeName : typeNames) {
+          typeNameQNameMap.put(typeName, namespace);
+        }
+      }
     }
   }
 
   public synchronized void unbind(ServiceReference<QueryFilterTransformer> reference) {
     if (reference == null) {
       return;
     }
-
     try {
       List<QName> namespaces = getNamespaces(reference);
-
       for (QName namespace : namespaces) {
         queryFilterTransformerMap.remove(namespace);
+        List<String> typeNames = getTypeNames(reference);
+        if (CollectionUtils.isNotEmpty(typeNames)) {
+          for (String typeName : typeNames) {
+            typeNameQNameMap.remove(typeName, namespace);
+          }
+        }
       }
     } finally {
       getBundleContext().ungetService(reference);
     }
   }
 
+  public synchronized Optional<QueryFilterTransformer> getTransformer(String typeName) {
+    if (StringUtils.isEmpty(typeName)) {
```

Comment:
❓ Can `typeName` be `null`? If yes then it should be tagged with `@Nullable` if not then there is no need to use `StringUtils.isEmpty()` simply call `typeName.isEmpty()`

Reply:
It can be null; added `@Nullable`